<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>SCENIC for TF Activation Scoring • dominoSignal</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.7.1/jquery.min.js" integrity="sha512-v2CJ7UaYy4JwqLDIrZUI/4hqeoQieOmAZNXBeQyjo21dadnwR+8ZaIJVT8EE2iyI61OV8e6M8PP2/4hpQINQ/g==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="SCENIC for TF Activation Scoring">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">


    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">dominoSignal</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/dominoSignal.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles

    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/cellphonedb_vignette.html">Using the CellPhoneDB Database</a>
    </li>
    <li>
      <a href="../articles/domino_object_vignette.html">Interacting with domino Objects</a>
    </li>
    <li>
      <a href="../articles/plotting_vignette.html">Plotting Functions and Options</a>
    </li>
    <li>
      <a href="../articles/tf_scenic_vignette.html">SCENIC for TF Activation Scoring</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/FertigLab/dominoSignal/" class="external-link">
    <span class="fab fa-github fa-lg"></span>

  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->



      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>SCENIC for TF Activation Scoring</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/FertigLab/dominoSignal/blob/master/vignettes/articles/tf_scenic_vignette.Rmd" class="external-link"><code>vignettes/articles/tf_scenic_vignette.Rmd</code></a></small>
      <div class="hidden name"><code>tf_scenic_vignette.Rmd</code></div>

    </div>

    
    
<p><a href="https://scenic.aertslab.org/" class="external-link">SCENIC</a> is our preferred
software for TF activity scoring. We recommend using the Python
implementation (<a href="https://pyscenic.readthedocs.io/en/latest/" class="external-link">pySCENIC</a>) as it is
faster than the original R implementation. Be aware, the use of <a href="https://scenic.aertslab.org/" class="external-link">SCENIC</a> is often the slowest and
most memory intensive step of this analysis pipeline. <a href="https://scenic.aertslab.org/" class="external-link">SCENIC</a> should be run on
computing resources with access to multi-core processing and large
amounts of memory.</p>
<div class="section level2">
<h2 id="file-downloads-for-scenic">File Downloads for SCENIC<a class="anchor" aria-label="anchor" href="#file-downloads-for-scenic"></a>
</h2>
<p>For this tutorial, <a href="https://pyscenic.readthedocs.io/en/latest/" class="external-link">pySCENIC</a> is used
as the method for TF activity inference. This requires downloading some
files prior to use. Here, we show how to download these files and the
way we run pySCENIC to generate the necessary files for use in
dominoSignal analyis.</p>
<p>A singularity image of <a href="https://scenic.aertslab.org/" class="external-link">SCENIC</a> v0.12.1 can be installed
from the DockerHub image as a source. <a href="https://scenic.aertslab.org/" class="external-link">SCENIC</a> requires a list of TFs,
motif annotations, and cisTarget motifs which are all available from the
authors of <a href="https://scenic.aertslab.org/" class="external-link">SCENIC</a> for human
(HGNC), mouse (MGI), and fly. The following will download everything
necessary for an analysis of a data set with HGNC gene labels for the
hg38 genome.</p>
<div class="section level3">
<h3 id="reference-files">Reference files<a class="anchor" aria-label="anchor" href="#reference-files"></a>
</h3>
<p>First we set up a temporary directory to store the files.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">temp_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempdir</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Then we build the singularity image and download reference files to
the temporary directory.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="va">SCENIC_DIR</span><span class="op">=</span><span class="st">"'temp_dir'/scenic"</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">mkdir</span> <span class="va">${SCENIC_DIR}</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a><span class="co"># Build singularity image</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="ex">singularity</span> build <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/aertslab-pyscenic-0.12.1.sif"</span> docker://aertslab/pyscenic:0.12.1</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a><span class="co"># Matrix containing motifs as rows and genes as columns and ranking position for each gene and motif (based on CRM scores) as values. URLs provided link to v2 feather files required for the 0.12.1 version of pySENIC.</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a><span class="ex">curl</span> <span class="st">"https://resources.aertslab.org/cistarget/databases/homo_sapiens/hg38/refseq_r80/mc_v10_clust/gene_based/hg38_10kbp_up_10kbp_down_full_tx_v10_clust.genes_vs_motifs.rankings.feather"</span> <span class="dt">\</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>  <span class="at">-o</span> <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/hg38_10kbp_up_10kbp_down_full_tx_v10_clust.genes_vs_motifs.rankings.feather"</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a><span class="ex">curl</span> <span class="st">"https://resources.aertslab.org/cistarget/databases/homo_sapiens/hg38/refseq_r80/mc_v10_clust/gene_based/hg38_500bp_up_100bp_down_full_tx_v10_clust.genes_vs_motifs.rankings.feather"</span> <span class="dt">\</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a>  <span class="at">-o</span> <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/hg38_500bp_up_100bp_down_full_tx_v10_clust.genes_vs_motifs.rankings.feather"</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" tabindex="-1"></a><span class="co"># List of genes encoding TFs in the HG38 reference genome</span></span>
<span id="cb2-16"><a href="#cb2-16" tabindex="-1"></a><span class="ex">curl</span> <span class="st">"https://resources.aertslab.org/cistarget/tf_lists/allTFs_hg38.txt"</span> <span class="dt">\</span></span>
<span id="cb2-17"><a href="#cb2-17" tabindex="-1"></a>  <span class="at">-o</span> <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/allTFs_hg38.txt"</span></span>
<span id="cb2-18"><a href="#cb2-18" tabindex="-1"></a></span>
<span id="cb2-19"><a href="#cb2-19" tabindex="-1"></a><span class="co"># Motif annotations based on the 2017 cisTarget motif collection. Use these files if you are using the mc9nr databases.</span></span>
<span id="cb2-20"><a href="#cb2-20" tabindex="-1"></a><span class="ex">curl</span> <span class="st">"https://resources.aertslab.org/cistarget/motif2tf/motifs-v9-nr.hgnc-m0.001-o0.0.tbl"</span> <span class="dt">\</span></span>
<span id="cb2-21"><a href="#cb2-21" tabindex="-1"></a>  <span class="at">-o</span> <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/motifs-v9-nr.hgnc-m0.001-o0.0.tbl"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="preparing-preprocessed-data">Preparing Preprocessed Data:<a class="anchor" aria-label="anchor" href="#preparing-preprocessed-data"></a>
</h3>
<p>dominoSignal was designed to be compatible with single cell
workflows, accepting gene by cell matrices for counts and scaled counts.
However, <a href="https://pyscenic.readthedocs.io/en/latest/" class="external-link">pySCENIC</a> is used
for TF activation inference, and it requires as input a cell by gene
matrix. This is the opposite orientation of many R based single cell
packages (such as <a href="https://satijalab.org/seurat" class="external-link">Seurat</a> or <a href="https://doi.org/doi:10.18129/B9.bioc.SingleCellExperiment" class="external-link">SingleCellExperiment</a>)
but is the default for Python based tools such as <a href="https://scanpy.readthedocs.io/en/stable/" class="external-link">scanpy</a>. The RNA
counts matrix can be saved as a tab-seperated value (.tsv) or
comma-sperated value (.csv) file after transposing the matrix to cell by
gene orientation. Below is an example of extracting the counts from a <a href="https://doi.org/doi:10.18129/B9.bioc.SingleCellExperiment" class="external-link">SingleCellExperiment</a>
object and saving it as a .tsv file.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pbmc_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">pbmc</span>, <span class="st">"counts"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.table</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="va">pbmc_counts</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">input_dir</span>, <span class="st">"/pbmc3k_counts.tsv"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">"\t"</span>,</span>
<span>    col.names <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></span></code></pre></div>
<p>tsv and csv files are very inefficient for storing large matrices. As
an alternative, the counts matrix can be saved as a loom file and
directly passed to <a href="https://pyscenic.readthedocs.io/en/latest/" class="external-link">pySCENIC</a> in this
format. Generating a loom file in R requires use of the <a href="https://github.com/mojaveazure/loomR" class="external-link">loomR</a> package. The
package automatically handles the transposition of the counts matrix to
cell by gene orientation as well. We recommend this approach to passing
a counts matrix to pySCENIC. However, users should be aware that <a href="https://github.com/mojaveazure/loomR" class="external-link">loomR</a> is not hosted on
CRAN or Bioconductor at the time of this vignette’s creation.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">loomR</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># save loom counts matrix</span></span>
<span><span class="va">pbmc_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">pbmc</span>, <span class="st">"counts"</span><span class="op">)</span></span>
<span><span class="va">pbmc_loom</span> <span class="op">&lt;-</span> <span class="fu">loomR</span><span class="fu">::</span><span class="fu">create</span><span class="op">(</span>filename <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">input_dir</span>, <span class="st">"/pbmc3k_counts.loom"</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">pbmc_counts</span><span class="op">)</span></span>
<span><span class="co"># connection to the loom file must be closed to complete file creation</span></span>
<span><span class="va">pbmc_loom</span><span class="op">$</span><span class="fu">close_all</span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>We will use the <code>pbmc3k\_counts.loom</code> file in this
tutorial.</p>
</div>
</div>
<div class="section level2">
<h2 id="running-scenic">Running SCENIC<a class="anchor" aria-label="anchor" href="#running-scenic"></a>
</h2>
<p><a href="https://pyscenic.readthedocs.io/en/latest/" class="external-link">pySCENIC</a> is
initiated using bash scripting in the terminal. The analysis consists of
three steps to score genes for TF motif enrichment, construct TF
regulons consisting of genes targeted by the TFs, and arrive at AUC
scores for enrichment of regulon gene transcription within each
cell.</p>
<div class="section level3">
<h3 id="grn-construct-tf-modules">grn: construct TF-modules<a class="anchor" aria-label="anchor" href="#grn-construct-tf-modules"></a>
</h3>
<p>Co-expression modules are used to quantify gene-TF adjacencies.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="ex">singularity</span> exec <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/aertslab-pyscenic-0.12.1.sif"</span> pyscenic grn <span class="dt">\</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>    <span class="st">"pbmc3k_counts.loom"</span> <span class="dt">\</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>    <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/allTFs_hg38.txt"</span> <span class="dt">\</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>    <span class="at">-o</span> <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/pbmc_adj_3k.tsv"</span> <span class="dt">\</span></span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>    <span class="at">--num_workers</span> 6 <span class="dt">\</span></span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>    <span class="at">--seed</span> 123</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a></span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a><span class="co"># Arguments:</span></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>    <span class="co"># path to the loom file</span></span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>    <span class="co"># list of TFs</span></span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>    <span class="co"># output directory for the adjacency matrix</span></span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a>    <span class="co"># number of CPUs to use if multi-core processing is available</span></span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>    <span class="co"># specify a random seed for reproducibility</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="ctx-construct-tf-regulons-with-pruning-based-on-tf-motif-enrichment">ctx: construct TF regulons with pruning based on TF motif
enrichment<a class="anchor" aria-label="anchor" href="#ctx-construct-tf-regulons-with-pruning-based-on-tf-motif-enrichment"></a>
</h3>
<p>The rankings of genes based on enrichment of TF motifs to the
transcription start site (TSS) are considered in the construction of
regulons, where target genes in the TF-modules are removed if they lack
motifs proximal to the TSS where the TF may bind.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="ex">singularity</span> exec <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/aertslab-pyscenic-0.12.1.sif"</span> pyscenic ctx <span class="dt">\</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>    <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/pbmc_adj.tsv"</span> <span class="dt">\</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>    <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/hg38_10kbp_up_10kbp_down_full_tx_v10_clust.genes_vs_motifs.rankings.feather"</span> <span class="dt">\</span></span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>    <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/hg38_500bp_up_100bp_down_full_tx_v10_clust.genes_vs_motifs.rankings.feather"</span> <span class="dt">\</span></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>    <span class="at">--annotations_fname</span> <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/motifs-v9-nr.hgnc-m0.001-o0.0.tbl"</span> <span class="dt">\</span></span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>    <span class="at">--expression_mtx_fname</span> <span class="st">"pbmc3k_counts.loom"</span> <span class="dt">\</span></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>    <span class="at">--mode</span> <span class="st">"dask_multiprocessing"</span> <span class="dt">\</span></span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>    <span class="at">--output</span> <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/regulons_pbmc_3k.csv"</span> <span class="dt">\</span></span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>    <span class="at">--num_workers</span> 1                                                          </span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a></span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a><span class="co"># Arguments:</span></span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>    <span class="co"># adjacency matrix output from grn</span></span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a>    <span class="co"># target rankings of motif enrichment within 10 kb of TSS</span></span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a>    <span class="co"># target rankings of motif enrichment within 500 bp upstream and 100 bp downstream of TSS</span></span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a>    <span class="co"># TF motif features</span></span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a>    <span class="co"># counts matrix loom file</span></span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a>    <span class="co"># enable multi-core processing</span></span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a>    <span class="co"># output file of learned TF regulons</span></span>
<span id="cb6-19"><a href="#cb6-19" tabindex="-1"></a>    <span class="co"># number of CPU cores</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="aucell-calculate-tf-activity-scores">aucell: calculate TF activity scores<a class="anchor" aria-label="anchor" href="#aucell-calculate-tf-activity-scores"></a>
</h3>
<p>Enrichment of a regulon is measured as the <strong>A</strong>rea
<strong>U</strong>nder the recovery <strong>C</strong>urve (AUC) of the
genes that define this regulon.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="ex">singularity</span> exec <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/aertslab-pyscenic-0.12.1.sif"</span> pyscenic aucell <span class="dt">\</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>    <span class="st">"pbmc3k_counts.loom"</span> <span class="dt">\</span></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>    <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/regulons_pbmc_3k.csv"</span> <span class="dt">\</span></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>    <span class="at">-o</span> <span class="st">"</span><span class="va">${SCENIC_DIR}</span><span class="st">/auc_pbmc_3k.csv"</span></span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>    </span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a><span class="co"># Arguments:</span></span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>    <span class="co"># counts matrix loom file</span></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>    <span class="co"># regulon table output from ctx</span></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>    <span class="co"># cell x TF matrix of TF enrichment AUC values</span></span></code></pre></div>
<p>The csv files created in this analysis can be used as the TF
activation inputs in a dominoSignal analysis. Please see our <a href="https://fertiglab.github.io/dominoSignal/articles/dominoSignal" class="external-link">Getting
Started</a> page for more information on how to perform an analysis with
dominoSignal. For a brief tutorial on the download of the necessary
files for using CellPhoneDB, please see the <a href="https://fertiglab.github.io/dominoSignal/articles/articles/cellphonedb_vignette" class="external-link">Using
the CellPhoneDB database</a> vignette.</p>
<details><summary>
Vignette Build Information
</summary><p>Date last built and session information:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Sys.time.html" class="external-link">Sys.Date</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "2024-11-20"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">#&gt; Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">#&gt; Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Matrix products: default</span></span>
<span><span class="co">#&gt; BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">#&gt; LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; locale:</span></span>
<span><span class="co">#&gt;  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">#&gt;  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">#&gt;  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">#&gt; [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; time zone: UTC</span></span>
<span><span class="co">#&gt; tzcode source: system (glibc)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; attached base packages:</span></span>
<span><span class="co">#&gt; [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; loaded via a namespace (and not attached):</span></span>
<span><span class="co">#&gt;  [1] digest_0.6.37     desc_1.4.3        R6_2.5.1          fastmap_1.2.0    </span></span>
<span><span class="co">#&gt;  [5] xfun_0.49         cachem_1.1.0      knitr_1.49        htmltools_0.5.8.1</span></span>
<span><span class="co">#&gt;  [9] rmarkdown_2.29    lifecycle_1.0.4   cli_3.6.3         sass_0.4.9       </span></span>
<span><span class="co">#&gt; [13] pkgdown_2.1.1     textshaping_0.4.0 jquerylib_0.1.4   systemfonts_1.1.0</span></span>
<span><span class="co">#&gt; [17] compiler_4.4.2    tools_4.4.2       ragg_1.3.3        bslib_0.8.0      </span></span>
<span><span class="co">#&gt; [21] evaluate_1.0.1    yaml_2.3.10       formatR_1.14      jsonlite_1.8.9   </span></span>
<span><span class="co">#&gt; [25] rlang_1.1.4       fs_1.6.5          htmlwidgets_1.6.4</span></span></code></pre></div>
</details>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

      </div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Christopher Cherry, Jacob T Mitchell, Sushma Nagaraj, Kavita Krishnan, Dmitrijs Lvovs.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

      </footer>
</div>






  </body>
</html>
